docs/doxygen/migration/mqtt.txt

```diff
@@ -0,0 +1,127 @@
+/**
+@page mqtt_migration MQTT
+@brief How to migrate an MQTT application from v3 to 202009.00.
+
+The MQTT library has been redesigned in 202009.00 as a solely single-threaded library. New features such as persistent session support and QoS 2 publish delivery have also been added.
+
+The features of the MQTT libraries in v3 and 202009.00 are defined by the common MQTT 3.1.1 spec; therefore, the two versions are similar.
+
+@section migration_mqtt_removed Removed features
+@brief The following features were present in v3, but removed in 202009.00:
+- <u>Auto-reconnect</u> <br>
+  Version 3 has an auto-reconnect feature triggered through a call to `aws_iot_mqtt_yield`. In version 202009.00, `aws_iot_mqtt_yield` has been removed, as well as the auto-reconnect feature. <br>
+  <b>Workaround:</b> When a version 202009.00 API call does not return @ref MQTTSuccess, the application should re-establish
+  the underlying transport connection and then call @ref MQTT_Connect. The application should employ an exponential backoff strategy when re-establishing connections.
+- <u>Thread support</u> <br>
+  Version 3 has an `_ENABLE_THREAD_SUPPORT_` macro for multi-threading use cases. Version 202009.00 has been redesigned to use only a single execution thread, and is
+  not thread-safe.
+
+@section migration_mqtt_datatypes Data Types
+@brief The following table lists equivalent data types in v3 and 202009.00.
+
+<table>
+    <tr>
+        <th>Version 3</th>
+        <th>Version 202009.00</th>
+        <th>Notes</th>
+    </tr>
+    <tr>
+        <td>enum QoS</td>
+        <td>@ref MQTTQoS_t</td>
+        <td>MQTT Quality of service.</td>
+    </tr>
+    <tr>
+        <td>AWS_IoT_Client</td>
+        <td>@ref MQTTContext_t</td>
+        <td>MQTT connection handle.</td>
+    </tr>
+    <tr>
+        <td>IoT_Publish_Message_Params</td>
+        <td>@ref MQTTPublishInfo_t</td>
+        <td>Parameters of an MQTT publish.</td>
+    </tr>
+    <tr>
+        <td>IoT_MQTT_Will_Options</td>
+        <td>@ref MQTTPublishInfo_t</td>
+        <td>Optional Will parameter of an MQTT connect. </td>
+    </tr>
+    <tr>
+        <td>IoT_Client_Connect_Params</td>
+        <td>@ref MQTTConnectInfo_t</td>
+        <td>Parameter of an MQTT connect. </td>
+    </tr>
+    <tr>
+        <td>IoT_Client_Init_Params</td>
+        <td><i>None</i></td>
+        <td>The members of this struct in v3 handled setup of the network connection.<br>This is handled outside of the MQTT library in 202009.00.</td>
+    </tr>
+    <tr>
+        <td>pApplicationHandler_t</td>
+        <td>@ref MQTTEventCallback_t</td>
+        <td>Function pointer of application callback to receive incoming data.</td>
```

Comment:
Was the V3 SDK also providing ack packets to the `pApplicationHandler_t`. If not, then that difference can also be mentioned

Reply:
Not sure if that would be any helpful in this table.